firebase-firestore/src/test/java/com/google/firebase/firestore/local/SQLiteLocalStoreTest.java

```diff
@@ -366,4 +366,201 @@ public void testDeeplyNestedServerTimestamps() {
     }
     assertThat(error.get()).isNull();
   }
+
+  @Test
+  public void testIndexAutoCreationWorks() {
+    Query query = query("coll").filter(filter("matches", "==", true));
+    int targetId = allocateQuery(query);
+
+    enableIndexAutoCreation();
+    setMinCollectionSizeToAutoCreateIndex(0);
+
+    applyRemoteEvent(addedRemoteEvent(doc("coll/a", 10, map("matches", true)), targetId));
```

Comment:
Can we setup the testing docs differently for each test you added:

we should try to cover different types and filters, like strings, numbers, arrays and inequalities/array queries.

Does this make sense?